mass_mailing_unique/hooks.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# Copyright 2016 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import _
+from odoo.exceptions import ValidationError
+
+
+def pre_init_hook(cr):
+    """Make sure there are no duplicates before installing the module.
+
+    If you define a unique key in Odoo that cannot be applied, Odoo will log a
+    warning and install the module without that constraint. Since this module
+    is useless without those constraints, we check here if all will work before
+    installing, and provide a user-friendly message in case of failure.
+    """
+    errors = list()
+
+    # Search for duplicates in emails
+    cr.execute("""SELECT c.email, l.name, COUNT(c.id)
+                  FROM
+                    mail_mass_mailing_contact AS c
+                    INNER JOIN mail_mass_mailing_contact_list_rel AS cl
```

Comment:
All you need to check, if you follow my suggestion below, is to check there are no duplicate emails in `mail_mass_mailing_contact`. No joins needed.